src/Uri.php

```diff
@@ -582,7 +616,7 @@ private function filterScheme($scheme)
             throw new \InvalidArgumentException('Scheme must be a string');
         }
 
-        return strtolower($scheme);
+        return \strtr($scheme, 'ABCDEFGHIJKLMNOPQRSTUVWXYZ', 'abcdefghijklmnopqrstuvwxyz');
```

Comment:
This is irrelevant with https://wiki.php.net/rfc/strtolower-ascii and I don't think we should add workarounds for things that even PHP core changes and think is stupid and not meant to be used.
Also there isn't and test for this added. So I'm 👎

Reply:
I guess we can re-visit this in a few years, when we drop PHP less than 8.2.